core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -467,13 +474,48 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
    * @param rootEntityType entity type
    * @param sanitizedEntityName entity name
    * @throws KeeperException if there is an error while setting or creating the znode
+   * @throws ControllerMovedException if no controller is defined, or a KRaft controller is defined
    */
   def setOrCreateEntityConfigs(rootEntityType: String, sanitizedEntityName: String, config: Properties): Unit = {
+    val controllerZkVersionOpt: Option[Int] = if (!enableEntityConfigNoController) {
```

Comment:
As this flag is used to guard against 1) no controller and 2) kraft controller, the naming enableEntityConfig`NoController` is a bit unsuitable. How about `enableEntityConfigCheck`?

Reply:
The name "enableEntityConfigNoController" is meant to convey "Enable setting entity configs even when there is no controller". But even as I've been coding this i've mixed up the meaning more than once 😅 

How about: `requireEntityConfigControllerCheck`?